pkg/platform/authentication/auth.go

```diff
@@ -89,27 +90,27 @@ func ClientAuth() runtime.ClientAuthInfoWriter {
 
 // Reset clears the cache
 func Reset() {
-	logging.Debug("Resetting")
 	persist = nil
 }
 
 // New creates a new version of Auth
 func New(cfg Configurable) *Auth {
-	logging.Debug("New Auth")
-
 	defer profile.Measure("auth:New", time.Now())
 	auth := &Auth{
 		cfg: cfg,
 	}
 
-	if auth.AvailableAPIToken() != "" {
+	return auth
+}
+
+func (s *Auth) Sync() error {
```

Comment:
Please document this. Why is it here? When does it have to be put after a call to `New()`? When does it not have to be called after `New()`?

Reply:
This is something that was done implicitly with `New`. This change makes me think that we want to keep the `LegacyGet` around. If so then maybe we should rename it. We may also want to update the docstrings to illustrate that `Sync` could be necessary. Otherwise, we should get rid of `LegacyGet` and rely on `New` so users of this API don't have to think about calling `Sync` at all.